test/hotspot/jtreg/runtime/cds/appcds/cacheObject/ArchiveHeapTestClass.java

```diff
@@ -173,7 +173,7 @@ static void testDebugBuild() throws Exception {
     static void testProductBuild() throws Exception {
         OutputAnalyzer output;
 
-        output = dumpHelloOnly("-XX:ArchiveHeapTestClass=NoSuchClass");
+        output = dumpHelloOnly("-XX:-IgnoreUnrecognizedVMOptions", "-XX:ArchiveHeapTestClass=NoSuchClass");
         mustFail(output, "VM option 'ArchiveHeapTestClass' is develop and is available only in debug version of VM.");
     }
```

Comment:
This is a very odd way to test this. The usual thing would be to mark the test as needing vm.debug and not even caring about what happens in a product build. This isn't actually testing the flag it is testing whether declaring a flag as develop works - which isn't what this test is really supposed to be for (that would be a VMFlags test). But this change will fix it. Though I'm not sure where the `-XX:+ignoreUnrecognizedVMOptions` is coming from.

Reply:
-XX:+IgnoreUnrecognizedVMOptions can come from -vmoptions in jtreg command.

Reply:
The `-XX:ArchiveHeapTestClass` flag is for testing purposes only. It allows arbitrary Java heap objects to be archived. We use this flag to test CDS internals.

However, archiving arbitrary Java heap objects is not a supported feature. This test case makes sure that we don't enable this flag by mistake in the product build.